contracts/root/staking/CustomSupernetManager.sol

```diff
@@ -129,9 +134,31 @@ contract CustomSupernetManager is ICustomSupernetManager, Ownable2StepUpgradeabl
         return validators[validator_];
     }
 
+    /**
+     *
+     * @inheritdoc ICustomSupernetManager
+     */
+    function premine(uint256 amount) external {
+        if (address(_rootERC20Predicate) == address(0)) {
+            revert Unauthorized("CustomSupernetManager: UNDEFINED_ROOT_ERC20_PREDICATE");
+        }
+
+        IERC20 nativeTokenRoot = IERC20(_rootERC20Predicate.nativeTokenRoot());
+        if (address(nativeTokenRoot) == address(0)) {
+            revert Unauthorized("CustomSupernetManager: UNDEFINED_NATIVE_TOKEN_ROOT");
+        }
+        require(!_genesis.completed(), "CustomSupernetManager: GENESIS_SET_IS_ALREADY_FINALIZED");
+
+        nativeTokenRoot.safeTransferFrom(msg.sender, address(_rootERC20Predicate), amount);
+        _genesis.insert(msg.sender, 0, amount);
```

Comment:
Flip the `insert` and `safeTransferFrom` lines.

Reply:
Fixed https://github.com/0xPolygon/core-contracts/pull/411/commits/96f761f4880fac0381dfc56a47a7f05753546ab4